networkx/algorithms/tests/test_simple_paths.py

```diff
@@ -259,3 +259,151 @@ def test_bidirectional_dijkstra_no_path():
+                 sorted([len(path) for path in  nx.all_simple_paths(G, 1, 12)]))
+
+def test_bidirectional_shortest_simple_paths_directed():
+    G = nx.cycle_graph(7, create_using=nx.DiGraph())
+    paths = nx.bidirectional.all_shortest_paths(G, 0, 3)
+    assert_equal([path for path in paths], [[0, 1, 2, 3]])
+
+def test_bidirectional_Greg_Bernstein():
+    g1 = nx.Graph()
+    g1.add_nodes_from(["N0", "N1", "N2", "N3", "N4"])
+    g1.add_edge("N4", "N1", weight=10.0, capacity=50, name="L5")
+    g1.add_edge("N4", "N0", weight=7.0, capacity=40, name="L4")
+    g1.add_edge("N0", "N1", weight=10.0, capacity=45, name="L1")
+    g1.add_edge("N3", "N0", weight=10.0, capacity=50, name="L0")
+    g1.add_edge("N2", "N3", weight=12.0, capacity=30, name="L2")
+    g1.add_edge("N1", "N2", weight=15.0, capacity=42, name="L3")
+    solution = [['N1', 'N0', 'N3'], ['N1', 'N2', 'N3'], ['N1', 'N4', 'N0', 'N3']]
+    result = list(nx.bidirectional.all_simple_paths(g1, 'N1', 'N3'))
+    assert_equal(result, solution)
+
+def test_bidirectional_weighted_shortest_simple_path():
+    def cost_func(path):
+        return sum(G.edge[u][v]['weight'] for (u, v) in zip(path, path[1:]))
+    G = nx.complete_graph(5)
+    weight = {(u, v): random.randint(1, 100) for (u, v) in G.edges()}
+    nx.set_edge_attributes(G, 'weight', weight)
+    cost = 0
+    for path in nx.bidirectional.all_shortest_paths(G, 0, 3):
+        this_cost = cost_func(path)
+        assert_true(cost <= this_cost)
+        cost = this_cost
+
+def test_bidirectional_directed_weighted_shortest_simple_path():
+    def cost_func(path):
+        return sum(G.edge[u][v]['weight'] for (u, v) in zip(path, path[1:]))
+    G = nx.complete_graph(5)
+    G = G.to_directed()
+    weight = {(u, v): random.randint(1, 100) for (u, v) in G.edges()}
+    nx.set_edge_attributes(G, 'weight', weight)
+    cost = 0
+    for path in nx.bidirectional.all_shortest_paths(G, 0, 3):
+        this_cost = cost_func(path)
+        assert_true(cost <= this_cost)
+        cost = this_cost
+
+def test_bidirectional_weight_name():
+    G = nx.cycle_graph(7)
+    nx.set_edge_attributes(G, 'weight', 1)
+    nx.set_edge_attributes(G, 'foo', 1)
+    G.edge[1][2]['foo'] = 7
+    paths = list(nx.bidirectional.all_simple_paths(G, 0, 3))
+    solution = [[0, 1, 2, 3], [0, 6, 5, 4, 3]]
+    assert_equal(paths, solution)
+
+@raises(nx.NetworkXError)
+def test_bidirectional_ssp_source_missing():
+    G = nx.Graph()
+    G.add_path([1,2,3])
+    paths = list(nx.bidirectional.all_shortest_paths(G, 0, 3))
+
+@raises(nx.NetworkXError)
+def test_bidirectional_ssp_target_missing():
+    G = nx.Graph()
+    G.add_path([1,2,3])
+    paths = list(nx.bidirectional.all_shortest_paths(G, 1, 4))
+
+@raises(nx.NetworkXNotImplemented)
+def test_bidirectional_ssp_multigraph():
+    G = nx.MultiGraph()
+    G.add_path([1,2,3])
+    paths = list(nx.bidirectional.all_shortest_paths(G, 1, 4))
```

Comment:
This will raise a `NetworkXError`, not `NetworkXNotImplemented`.
```
======================================================================
ERROR: test_simple_paths.test_bidirectional_ssp_multigraph
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/nose/tools/nontrivial.py", line 60, in newfunc
    func(*arg, **kw)
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/networkx/algorithms/tests/test_simple_paths.py", line 409, in test_bidirectional_ssp_multigraph
    paths = list(nx.bidirectional.all_shortest_paths(G, 1, 4))
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/networkx/algorithms/bidirectional.py", line 284, in all_shortest_paths
    raise nx.NetworkXError('target node %s not in graph' % target)
NetworkXError: target node 4 not in graph
```